pkg/platform/provider/baremetal/preflight/checks.go

```diff
@@ -371,6 +390,33 @@ func (dac DirAvailableCheck) Check() (warnings, errorList []error) {
 	return nil, errorList
 }
 
+// Check the kernel version and kernel parameter for Cilium installation.
+type CiliumCheck struct {
```

Comment:
why create "CiliumCheck" rather than enhance existing "KernelCheck" and 'KernelModuleCheck'

Reply:
Because "KernelCheck" checks Kernel and 'KernelModuleCheck' checks the module.These methods are alll util method.So I want to make a new solid method to check the Kernel Parameter.Maybe I should rename it the KernelParameter check to makes it util.

Reply:
take RunMasterChecks as example, please

Reply:
`RunNodeChecks` need add logic to check node whether suitable for install cilium for `add new node` and `master scale up` case, please consolidate `RunNodeChecks` and `EnsureCilium` to avoid double check